client/app/views/popover_screens/guests.coffee

```diff
@@ -158,31 +183,79 @@ module.exports = class GuestPopoverScreen extends EventPopoverScreenView
     # Handle guest addition. `userInfo` is passed when called by the typeahead.
     onNewGuest: (userInfo = null) ->
 
+        # Autocomplete was used.
         if userInfo? and typeof(userInfo) is "string"
-            [email, contactID] = userInfo.split(';')
+            [channel, contactID] = userInfo.split(';')
+        # Field was entered manually.
         else
-            email = @$('input[name="guest-name"]').val()
+            channel   = @$('input[name="guest-name"]').val()
             contactID = null
 
-        # An empty value should not be submitted.
-        email = email.trim()
-        if email.length > 0
+        # Determine if guest's "channel" of communication is the url of his cozy
+        # or his mail address.
+        if (channel.indexOf "@") < 0
+            cozy  = channel
+        else
```

Comment:
Same thing here, I think relying on the presence of an `@`is really weak (a cozy URL should be formatted like something `https://gregory@gregserver.perso/cozy/`

But again, the best solution should be to have the contact's source (email or cozy) passed to the `onNewGuest` method somehow.

Maybe using an email regexp here should be a good compromise ? What do you think ?